rplugin/python3/denite/source/workspaceSymbol.py

```diff
@@ -17,28 +20,33 @@ def __init__(self, vim):
         self.name = 'workspaceSymbol'
         self.kind = 'file'
 
-    def convert_to_candidates(self, symbols: List[Dict]) -> List[Dict]:
-        candidates = []
-        pwd = self.vim.funcs.getcwd()
-        for sb in symbols:
-            name = sb["name"]
-            uri = sb["location"]["uri"]
-            filepath = uri_to_path(uri)
-            relpath = path.relpath(filepath, pwd)
-            start = sb["location"]["range"]["start"]
-            line = start["line"] + 1
-            character = start["character"] + 1
-            candidates.append({
-                "word": "{}:{}:{}:\t{}".format(
-                    relpath, line, character, name),
-                "action__path": filepath,
-                "action__line": line,
-                "action__col": character,
-            })
-
-        return candidates
+    def highlight(self):
+        highlight_setup(self, SYMBOL_CANDIDATE_HIGHLIGHT_SYNTAX)
 
     def gather_candidates(self, context):
+        context['is_interactive'] = True
+        prefix = context['input']
+        bufnr = context['bufnr']
+
+        cached_prefix = context.get('__cached_prefix')
+        if cached_prefix is not None and prefix.startswith(cached_prefix):
+            return context['__cached_candidates']
+
+        current_buffer = self.vim.current.buffer.number
+        if current_buffer != bufnr:
```

Comment:
Just curious, under which case would this becoming true?

Reply:
LanguageClient does everything in the context of the current buffer. When interactively filtering results the current buffer is the denite buffer not the source buffer. This means that LanguageClient keeps returning empty results because there is not server defined for denite buffers (obviously). This is a trick to open a new tab with the original buffer and then run the command in that buffer.

The proper way to fix is to have LanguageClient have APIs to run in the context of a buffer or a specific server, but it would have been much more complicated than this hack and I (and others) need this to work now as most projects have more than a 1000 symbols.

I will document this in the code and open an issue requesting these APIs.